### PR TITLE
[SPARK-9835] [PySpark] fix install pylint

### PR DESCRIPTION
**Pull request by davies**

### CODE REVIEW

**Comment by davies** [comment]
cc @yhuai

**Comment by JoshRosen** [comment]
Note that the code to actually run the PyLint checks is still commented out, so you might want to revert that commit as part of this PR if you're trying to re-enable those checks.

**Comment by davies** [comment]
@JoshRosen What's the context when we disable pylint checker? Should we just remove the pylint totally?

**Comment by JoshRosen** [comment]
@davies, it was reverted while triaging a build break today.  When the PyLint download started failing builds today we initially misdiagnosed the problem as an issue with the checks themselves and attempted to revert those checks first.

**Comment by davies** [comment]
@JoshRosen I see, we can fix the installer first, delay the decision later.

File: dev/lint-python [hunk 1]
```diff
@@ -58,21 +58,21 @@ export "PYTHONPATH=$SPARK_ROOT_DIR/dev/pylint"
...
-#         exit "$easy_install_status"
-#     fi
-#
-#     rm "$PYLINT_INSTALL_INFO"
-#
-# fi
+if [ ! -d "$PYLINT_HOME" ]; then
+    mkdir "$PYLINT_HOME"
+    # Redirect the annoying pylint installation output.
+    easy_install -d "$PYLINT_HOME" pylint==1.4.4 >> "$PYLINT_INSTALL_INFO" 2>&1
```
Comment (yhuai) [inline dev/lint-python]: Ah, I see. The problem is indeed the redirect. Why our PR builder was fine with that PR? We did not check python style?

**Comment by yhuai** [comment]
btw, seems https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1221/consoleFull did not trigger python style check?

File: dev/lint-python (hunk 1 above)
Comment (davies) [inline dev/lint-python]: It only fail for the first time

**Comment by davies** [comment]
pylint is not that useful and is disabled now, close this.

**Pull request closed without merging**